T2D-53 Modified file tracking service.

jorizci opened the pull request:
Hash calculus is postponed until the end of the add function. Now when a new file is added a temptative temporary entry is created with the information to access the file. Then the hash has been calculated properly this registry is updated.

jorizci committed:
Modified upload method to do file a lazy hash calculus.

cyenyxe commented on src/main/java/uk/ac/ebi/ampt2d/services/FileTrackingService.java:
```diff
@@ -44,29 +44,42 @@
         String fileHash = null;
+        FileMetadata storedFileMetadata = null;
         try {
+            storedFileMetadata = storeTemporalFileMetadata(sourceFilePath);
```
"temporary" instead of "temporal"

cyenyxe commented on src/main/java/uk/ac/ebi/ampt2d/services/FileTrackingService.java:
```diff
@@ -44,29 +44,42 @@
     }
 
+    private void deleteFileMetadata(FileMetadata storedFileMetadata) {
+        for(SourceFilePath sourceFilePath: storedFileMetadata.getSourceFilePaths()){
```
White-space formatting.

cyenyxe commented on src/main/java/uk/ac/ebi/ampt2d/services/FileTrackingService.java:
```diff
@@ -44,29 +44,42 @@
-        } catch (IOException e){
-            fileSystemStorageService.delete(sourceFilePath);
+            storedFileMetadata.setHash(fileHash);
+            return fileMetadataRepository.save(storedFileMetadata);
```
Maybe these 2 lines could be extracted to a single method like `updateFileMetadataHash`? That would make even more straightforward to follow the 3-step process:

1. Store temporary file metadata
2. Calculate file hash
3. Update file metadata hash

jorizci committed:
Applied pr suggestions.
-Extracted hash update to a private function
-Changed temporal to temporary
-Checked white space indentation
-Added small comments
